src/playsim/p_actionfunctions.cpp

```diff
@@ -2458,13 +2459,14 @@ DEFINE_ACTION_FUNCTION(AActor, CheckIfTargetInLOS)
 	{
 		ACTION_RETURN_BOOL(false);
 	}
-	double distance = self->Distance3D(target);
+	double distance_squared = self->Distance3DSquared(target);
 
-	if (dist_max && (distance > dist_max))
+	if (dist_max && (dist_max >= 0 && distance_squared > (dist_max * dist_max) ) )
```

Comment:
This comparison is wrong.

```suggestion
	if (dist_max && (dist_max < 0 || distance_squared > dist_max*dist_max))
```

Reply:
Ehhhh....but its literally same thing as before, just in another form.

Reply:
Wrong: if `dist_max` is negative, then the condition in your code returns `false`, which is not what the original code did (in that case it returns `true`). You need to preserve the original behavior here. The same is said for the other comments.